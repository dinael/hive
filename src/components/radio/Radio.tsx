import React, { FC, useState } from 'react'

export type InputRadioProps = {
  value: string
  id?: string,
  onChange: (value: string) => void
  label: string
  required?: boolean
  error?: boolean
  errorText?: string
  className?: string
}

export const InputRadio: FC<InputRadioProps> = ({
  value,
  id,
  onChange,
  label,
  required = false,
  error,
  errorText,
  className,
  ...props
}) => {

  const [isSelected, setTouched] = useState(false)
  const [hasError, setError] = useState(false)

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value)
  }

  const handleBlur = () => {
    setTouched(true)
  }

  return (
    <label
      htmlFor={id}
      className={`${className ? className : ''}`}
      {...props}>
      <span>{label}</span>
      <input
        id={id}
        type="radio"
        name="inputRadio"
        value={value}
        checked={value === value}
        onChange={handleChange}
        onBlur={handleBlur}
        required={required}
        aria-describedby={error ? `validation-message-${id}` : ''}
      />
      {error ?
        <span
          role="alert"
          aria-live="assertive"
          aria-atomic="true"
          id={`validation-message-${id}`}>
          {errorText}
        </span> : null}
    </label>
  )
}

export default InputRadio
