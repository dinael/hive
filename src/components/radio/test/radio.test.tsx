import { describe, test, expect, vi } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import { InputRadio, InputRadioProps } from '../Radio'


describe('InputRadio', () => {
  const defaultProps: InputRadioProps = {
    value: 'example-value',
    label: 'Example label:',
    error: true,
    id: 'radio-test',
    errorText: 'Error text',
    onChange: vi.fn(),
  }

  test('InputRadio renders snapshot', () => {
    const { container } = render(<InputRadio {...defaultProps}/>)
    expect(container).toMatchSnapshot()
  })

  test('InputRadio renders with the correct label', () => {
    const { container } = render(<InputRadio {...defaultProps}/>)
    const LabelElement = container.querySelector('label > span')
    expect(LabelElement?.textContent).toContain('Example label')
    expect(LabelElement?.className).toContain('label-text-component')
  })

  test('InputRadio renders with the correct value', () => {
    const { container } = render(<InputRadio {...defaultProps} />)
    const inputElement = container.querySelector('input')
    expect(inputElement?.value).toBe('example-value')
  })

  test('InputRadio calls onChange handler when clicked', () => {
    const onChangeMock = vi.fn()
    const { container } = render(<InputRadio {...defaultProps} onChange={onChangeMock} />)
    const inputElement = container.querySelector('input')
    if (inputElement) {
      fireEvent.change(inputElement, { target: { value: 'example-value' } })
    }
    expect(onChangeMock).toHaveBeenCalledTimes(0)
  })

  test('InputRadio calls onChange handler when clicked', () => {
    const onChangeMock = vi.fn()
    const { container } = render(<InputRadio {...defaultProps} onChange={onChangeMock} />)
    const inputElement = container.querySelector('input')
    if (inputElement) {
      fireEvent.change(inputElement, { target: { value: 'example-value' } })
    }
    expect(inputElement?.value).toBe('example-value')
  })

})
