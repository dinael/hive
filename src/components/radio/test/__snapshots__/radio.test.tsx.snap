// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputRadio > InputRadio renders snapshot 1`] = `
<div>
  <label
    class=""
    for="radio-test"
  >
    <span
      class="
    label-text-component
    
    
    
    
  "
    >
      Example label::
    </span>
    <input
      aria-describedby="validation-message-radio-test"
      id="radio-test"
      name="inputRadio"
      type="radio"
      value="example-value"
    />
    <span
      aria-atomic="true"
      aria-live="assertive"
      id="validation-message-radio-test"
      role="alert"
    >
      Error text
    </span>
  </label>
</div>
`;
